addons/mail_group/tests/common.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command
+from odoo.addons.mail.tests.common import mail_new_test_user
+from odoo.addons.test_mail.tests.common import TestMailCommon
+
+
+class TestMailListCommon(TestMailCommon):
+    def setUp(self):
+        super(TestMailListCommon, self).setUp()
+        self.group = self.env['mail.group'].create({
+            'name': 'Test group',
+            'alias_name': 'group',
+            'moderation': True,
+            'moderator_ids': [Command.link(self.user_employee.id)],
+        })
+        self.group_2 = self.env['mail.group'].create({
+            'name': 'group 2',
+            'alias_name': 'group_2',
+            'moderation': True,
+            'moderator_ids': [Command.link(self.user_admin.id)],
+        })
+        self.moderation = self.env['mail.group.moderation'].create({
+            'mail_group_id': self.group.id,
+            'email': 'banned_member@example.com',
+            'status': 'ban',
+        })
+
+        self.member_1 = self.env['mail.group.member'].create({
+            'email': '"Member 1" <member_1@example.com>',
+            'mail_group_id': self.group.id,
+        })
+        self.member_2 = self.env['mail.group.member'].create({
+            'email': 'member_2@example.com',
+            'mail_group_id': self.group.id,
+        })
+        self.banned_member = self.env['mail.group.member'].create({
+            'email': 'banned_member@example.com',
+            'mail_group_id': self.group.id,
+        })
+        self.member_partner = self.env['mail.group.member'].create({
+            'partner_id': self.partner_employee.id,
+            'email': self.partner_employee.email,
+            'mail_group_id': self.group.id,
+        })
+        self.member_group_2 = self.env['mail.group.member'].create({
+            'email': '"Member group 2" <member_group_2@example.com>',
+            'mail_group_id': self.group_2.id,
+        })
+
+        self.env["ir.config_parameter"].sudo().set_param("mail.catchall.domain", "example.com")
+
+        self.user_without_email = mail_new_test_user(
+            self.env, login='user_test_mail_group',
+            email=False,
+            groups='base.group_user',
+            company_id=self.company_admin.id,
+            name='User without email',
+        )
+
+        self.user_employee_2 = mail_new_test_user(
+            self.env, login='employee_2',
+            email='employee_2@example.com',
+            groups='base.group_user',
+            company_id=self.company_admin.id,
+            name='User without email',
+        )
+
+        self.assertTrue(self.user_admin.has_group('mail_group.group_mailing_list_administrator'))
```

Comment:
Strange to have an assert in a setup. Maybe simply add the group to admin ?

Reply:
Not really needed, I can drop it